core/mixins.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from datetime import datetime
+from app import db
+
+# Creating the customer model
+class BaseModelMixen(object):
+    id = db.Column(db.Integer, primary_key=True)
+    date_created = db.Column(db.DateTime, default=datetime.utcnow)
+    # Nullable default is true, Unique default is False
+
+    def save(self):
+        # Used to save data
+        db.session.add(self)
```

Comment:
a golden rule of inheritance is that model methods and attributes always flow down to children, never up to parents.

here, you've defined `BaseModelMixen` to not inherit from `db.Model` - that means that these methods (e.g. `save` and `delete`) call a method on the `db` instance that's not a part of thise model's definition.

consider how you would use this mixin to define future child models later.